core/serializers.py

```diff
@@ -1,8 +1,25 @@
-from .models import Hackathon
 from rest_framework import serializers
 from authentication.models import User
 
+from .models import Hackathon, Team
+
+class TeamSerializer(serializers.ModelSerializer):
+    class Meta:
+        model = Team
+        fields = '__all__'
+
+class TeamCreateSerializer(serializers.ModelSerializer):
+
+    def validate(self, data):
+        hackathon_id = self.context.get('view').kwargs['pk']
+        if hackathon_id != data['hackathon']:
+            raise serializers.ValidationError("Mismatch of Hackathon ID in url and request body!")
+        return data
+    class Meta:
+        model = Team
+        fields = '__all__'
```

Comment:
I think for creating a team you need only the team name in the request body.

Reply:
And what about the member details, leader etc?

Reply:
That can be deduced from the token auth, right? And members join afterward by entering the team code.

Reply:
Ah, alright. So, initally when team is created, we would set the logged in user as the Leader and member. And set a TeamID, and register the team in the hackathon.

Now another user can enter TeamID to join the team. And for that we'll have to create another endpoint. Should I create that endpoint too or just proceed with the initial team creation?